### DISCUSSION TITLE
Docker setup problems

### DISCUSSION THREAD

**Discussion by jabarragann** [surgical-robotics-ai/surgical_robotics_challenge#52] [General]
Hi all, 

I have been trying to setup up the docker containers for the challenge submission, but I have been facing some problems with the network configuration. The main issue is that the ROS inside the docker container cannot connect to the Roscore running outside.

After setting up my container I tried running the command `rostopic list` which produced the following error.

```ERROR: Unable to communicate with master!```

Googling a little bit on the issue, I found that `export ROS_MASTER_URI=http://10.0.0.8:44403/` could help fix the problem. However, this also leads to an error when calling `rostopic list`.

```    
return u.close()
  File "/usr/lib/python3.8/xmlrpc/client.py", line 655, in close
    raise Fault(**self._stack[0])
xmlrpc.client.Fault: <Fault 1: '<class \'Exception\'>:method "getSystemState" is not supported'>
```

Any pointers on how to solve the problem would be appreciated!

**Comment by adnanmunawar** [comment]
Hello Juan,

Discussing with @pkazanzides, we decided to make it easy for the challenge participants to make submissions. The participants can just upload their code to GitHub with instructions on how to run it and we shall take care of the rest. We are about to announce this on the challenge website.

And I shall get back to the Docker-ROS question in a bit for future reference.

**Reply by jabarragann** [reply]
That is great Adnan! Let me know about the Docker-ROS issue whenever you get some time.

**Comment by jabarragann** [comment]
Hi Adnan, 

I wanted to let you know that I finally was able to solve the docker issues, a little bit too late but I will leave the solution here just in case it is useful for anybody else.

The problem was that I initially installed docker-desktop instead of docker engine. Apparently, docker-desktop does not allow communication between the host and the containers at least out of the box. After uninstalling docker-desktop and installing docker-engine everything worked fine. I added the instructions to get my docker image in my GitHub readme.

Lastly, I just realized that I always used the simulation without the needle's thread, so probably my algorithms will not work as well when adding the thread. So, if possible it will be great if you can test my code on the simulation without the needle's thread.

Thanks!
